pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -82,6 +82,9 @@ func (s *Service) ReconcileLoadBalancer(openStackCluster *infrav1.OpenStackClust
 
 	lb, err := s.getOrCreateAPILoadBalancer(openStackCluster, clusterResourceName)
 	if err != nil {
+		if errors.Is(capoerrors.ErrFilterMatch, err) {
```

Comment:
Should this be the other way around?
```suggestion
		if errors.Is(err, capoerrors.ErrFilterMatch) {
```